vault/logical_system_paths.go

```diff
@@ -2093,6 +2093,126 @@ func (b *SystemBackend) pluginsReloadPath() *framework.Path {
 	}
 }
 
+func (b *SystemBackend) pluginsRuntimesCatalogCRUDPath() *framework.Path {
+	return &framework.Path{
+		Pattern: "plugins/runtimes/catalog(/(?P<type>container))?/(?P<name>.+)",
```

Comment:
Separately, a couple of suggestions on the path. I don't think the `/container` segment should be optional. I'm also slightly more inclined to keep everything under the `/v1/sys/plugins/catalog` prefix rather than `/v1/sys/plugins/runtimes/catalog` although I don't have any well thought through reasoning on that one, so I'm open to thoughts.

```suggestion
		Pattern: "plugins/catalog/runtimes/(?P<type>container)/(?P<name>.+)",
```

Reply:
My thought of going with `plugins/runtimes/catalog` was `runtimes` has its own catalog, and it's not stored under `plugins` catalog. `plugins/catalog/runtimes/` seems to indicate that `runtimes` as a plugin type, and the runtime catalog is stored under the plugin catalog. I think I might go by the book too much here by matching the view and model. Let me know your thoughts.

Reply:
On reflection, I agree - I think stick with what you've got. While we technically could carve out the space from `/v1/sys/plugins/catalog/runtimes`, it's somewhat more likely to overlap with existing policies, and it does seem like a clearer separation to use `/v1/sys/plugins/runtimes/catalog` 👍